frontend/public/components/secret.tsx

```diff
@@ -258,13 +229,57 @@ const SecretsPage = (props) => {
   );
 };
 
-const SecretsDetailsPage = (props) => (
-  <DetailsPage
-    {...props}
-    buttonActions={actionButtons}
-    menuActions={menuActions}
-    pages={[navFactory.details(detailsPage(SecretDetails)), navFactory.editYaml()]}
-  />
-);
+const SecretsDetailsPage: React.FCC<SecretDetailsPageProps> = (props) => {
+  const { t } = useTranslation();
+  const { name: secretName, namespace, kindObj: kind } = props;
+
+  const addSecretToWorkloadLauncher = useAddSecretToWorkloadModalLauncher({
+    secretName,
+    namespace,
+  });
+
+  const actionButtons = React.useMemo(
+    () => [
+      () => ({
+        callback: () => addSecretToWorkloadLauncher(),
+        label: t('public~Add Secret to workload'),
+      }),
+    ],
+    [t, addSecretToWorkloadLauncher],
+  );
+
+  const customActionMenu = (kindObj: K8sResourceKind, obj: K8sModel) => {
+    const resourceKind = referenceFor(kindObj);
+    const context = { [resourceKind]: obj };
+    return (
+      <ActionServiceProvider context={context}>
+        {({ actions, options, loaded }) =>
+          loaded && (
+            <ActionMenu actions={actions} options={options} variant={ActionMenuVariant.DROPDOWN} />
+          )
+        }
+      </ActionServiceProvider>
+    );
+  };
+
+  return (
+    <DetailsPage
+      {...props}
+      kind={kind.kind}
+      buttonActions={actionButtons}
+      customActionMenu={customActionMenu}
```

Comment:
Working on `subscription.tsx` and realized this can be a lot simpler

```suggestion
      customActionMenu={(kindObj: K8sModel, obj: K8sResourceKind) => (
        <LazyActionMenu context={{ [referenceFor(kindObj)]: obj }} />
      )}
```